cmd/terramate/cli/cli.go

```diff
@@ -294,40 +305,27 @@ func (c *cli) initStack(dirs []string) error {
 	return nil
 }
 
-func (c *cli) listStacks(
-	basedir string,
-	mgr *terramate.Manager,
-	isChanged bool,
-) ([]terramate.Entry, error) {
-
+func (c *cli) listStacks(mgr *terramate.Manager, isChanged bool) ([]terramate.Entry, error) {
 	if isChanged {
-		git, err := newGit(basedir, c.inheritEnv, true)
-		if err != nil {
-			return nil, err
-		}
-		if err := c.checkDefaultRemote(git); err != nil {
-			return nil, err
-		}
-		if err := c.checkLocalDefaultIsUpdated(git); err != nil {
-			return nil, err
-		}
 		return mgr.ListChanged()
 	}
 
 	return mgr.List()
 }
 
-func (c *cli) printStacks(basedir string) error {
-	mgr := terramate.NewManager(basedir, c.baseRef)
-	entries, err := c.listStacks(basedir, mgr, c.parsedArgs.List.Changed)
+func (c *cli) printStacks() error {
+	mgr := terramate.NewManager(c.root(), c.prj.baseRef)
+	entries, err := c.listStacks(mgr, c.parsedArgs.Changed)
 	if err != nil {
 		return err
 	}
 
-	trimPart := c.wd + string(os.PathSeparator)
 	for _, entry := range entries {
 		stack := entry.Stack
-		stackdir := strings.TrimPrefix(stack.Dir, trimPart)
+		stackdir, ok := c.showdir(stack.Dir)
```

Comment:
Just to give some feedback from the perspective of someone new to the code, I haven't got to the definition of showdir yet, but in this context it is kinda hard to intuitively understand what is going on, what would "showdir" do/check I wonder :thinking: